src/nodedef.h

```diff
@@ -819,6 +831,11 @@ class NodeDefManager {
 	 * Even constant NodeDefManager instances can register listeners.
 	 */
 	mutable std::vector<NodeResolver *> m_pending_resolve_callbacks;
+
+	/*!
+	 * Fast cache of content lighting flags.
+	 */
+	ContentLightingFlags m_content_lighting_flag_cache[CONTENT_MAX + 1L];
```

Comment:
It is not necessary to waste RAM here. You could move these flags to `ContentFeatures` and implement an inline `NodeDefManager::getNoCheck(content_t id)` function to keep the lookup speed.